### PR TITLE
feat(module:calendar): sync some changes from ant-design (#2769)

### PR DESCRIPTION
**Pull request by zhytang**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #2769 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/calendar/nz-calendar.spec.ts [hunk 1]
```diff
@@ -466,3 +466,18 @@ class NzTestCalendarMonthCellComponent { }
   `
 })
 class NzTestCalendarMonthFullCellComponent { }
+
```
Comment (vthinkxie) [inline components/calendar/nz-calendar.spec.ts]: test needed

**Comment by zhytang** [comment]
add test

**Comment by vthinkxie** [comment]
Hi @zhytang 
plz rebase your pr onto master, thanks a lot

**Comment by zhytang** [comment]
> Hi @zhytang
> plz rebase your pr onto master, thanks a lot

Sorry, I have rebase, but there is a mistake, and a new pr (2963)

**Pull request closed without merging**